src/components/movies/MovieList.vue

```diff
@@ -19,7 +20,54 @@ export default {
   components: {
     MovieCard,
   },
+  data() {
+    return {
+      selectedGenre: "",
+    };
+  },
+  props: {
+    search: {
+      type: String,
+      default: "",
+    },
+  },
+  computed: {
+    searchedMovies() {
+      let re = new RegExp(this.search, "i");
+      return this.$store.getters.movieList.filter((movie) =>
+        movie.title.match(re)
+      );
+    },
+    moviesByCategory() {
+      const moviesByCategory = this.searchedMovies.filter(
+        (movie) => movie.genre.name === this.$store.getters.selectedGenre
+      );
+      return this.$store.getters.selectedGenre == "All Movies" ||
+        this.$store.getters.selectedGenre == ""
+        ? this.searchedMovies
+        : moviesByCategory;
+    },
+  },
```

Comment:
I have a cool idea and perhaps you might like it too 💡 
Here we essentially filter movies we get from our vuex store, by a selected genre that is... stored in vuex store too.
This is an excellent case to introduce a getter here in the store and filter those movies there!
We would only need to use `mapGetters` or access the state here to get those movies then but it would totally work and allow us to potentially access the filtered movies from other places 🚀
Getters are an excellent place for filtering or transforming data we store!

Reply:
I really love the idea of being able to move all this logic to the store <3 but I am struggling to understand how to do it....
moviesByCategory depends on another computed in this component searchedMovies
searchedMovies relies on local data from search input that it recieves by prop

To be able to move the moviesByCategory to store would I have to move all depending function and data above to the store as well?

Reply:
So, my idea is - let's have a getter for movies filtered by genre for a start, so essentially something more or less like:
```
getters: {
  moviesForSelectedGenre: (state) => {
    if (state.selectedGenre === "All Movies" || state.selectedGenre === "") {
      return state.moviesRecord; // return all of them
    }
    
    // otherwise return movies for selected genre
    return state.moviesRecord.filter(movie => movie.genre.name === state.selectedGenre);
  },
  ...
}
```
Then we could use this getter here to *further* filter that list by the regexp like you did before :)
```
let re = new RegExp(this.search, "i");
return this.moviesForSelectedGenre.filter((movie) =>
  movie.title.match(re)
);
```
So now we have all that we can done in the store for convenience (filter by category), and all extra filtering done here by the input value 😄

Reply:
THANK YOU x1000000 FOR THIS ESSAY and extremely detailed explanation ( aka actual solution :) )